### PR TITLE
fix: reset usdt approval for mainnet only

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Ensures we only trigger approval resets for mainnet, since implementations other than the original USDT one are not affected by this.

## Issue (if applicable)

- contributes to https://github.com/shapeshift/web/issues/7468

## Risk

> High Risk PRs Require 2 approvals

Low

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing

- Ensure USDT approval resets are still triggered for mainnet USDT when required
- Ensure USDT approval resets are never triggered for USDT flavors other than mainnet

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

## Screenshots (if applicable)

Tested against develop (right)

- Mainnet USDT reset required

image
image

- Other chains - USDT reset not required

image
image

### CODE REVIEW

File: src/hooks/queries/useIsApprovalRequired.ts [hunk 1]
```diff
@@ -15,16 +12,14 @@ type UseIsApprovalRequired = {
   amountCryptoBaseUnit: string
 }
 
+export const usdtAssetId: AssetId = 'eip155:1/erc20:0xdac17f958d2ee523a2206206994597c13d831ec7'
```
Comment (gomesalexandre) [inline src/hooks/queries/useIsApprovalRequired.ts:15]:
```suggestion
const usdtAssetId: AssetId = 'eip155:1/erc20:0xdac17f958d2ee523a2206206994597c13d831ec7'
```

File: src/hooks/queries/useIsApprovalRequired.ts [hunk 2]
```diff
@@ -15,16 +12,14 @@ type UseIsApprovalRequired = {
   amountCryptoBaseUnit: string
 }
 
+const usdtAssetId: AssetId = 'eip155:1/erc20:0xdac17f958d2ee523a2206206994597c13d831ec7'
```
Comment (0xApotheosis) [inline src/hooks/queries/useIsApprovalRequired.ts:15]: Nitpick on naming, should this be `usdtOnEthereumAssetId`?

**Review by 0xApotheosis — APPROVED** [review]
✅ USDT on Ethereum correctly uses the reset flow when necessary

Screenshot 2024-09-02 at 7 34 08 PM

✅ USDT on other chains does not show the reset flow, even when there is an existing, insufficient approval on chain

Screenshot 2024-09-02 at 7 35 19 PM

File: src/hooks/queries/useIsApprovalRequired.ts (hunk 2 above)
Comment (gomesalexandre) [reply src/hooks/queries/useIsApprovalRequired.ts:15]: Conventionally, we never use any chain terminology for mainnet flavours! Note I've moved this to caip in https://github.com/shapeshift/web/pull/7670/files#diff-07cb24655c483f16cbe3aee6ae57352da77b488cda993f70567297dc1e1fe308R28, happy to change the terminology across the board there if we find it easier to understand.

File: src/hooks/queries/useIsApprovalRequired.ts [hunk 3]
```diff
@@ -35,17 +30,14 @@ export const useIsApprovalRequired = ({
   }, [allowanceCryptoBaseUnitResult, amountCryptoBaseUnit])
 
   const isAllowanceResetRequired = useMemo(() => {
+    if (!assetId) return
     if (!allowanceCryptoBaseUnitResult.data) return
+
+    if (fromAssetId(assetId).chainId !== ethChainId) return false
     const hasAllowance = bnOrZero(allowanceCryptoBaseUnitResult.data).gt(0)
-    const isUsdt = usdtAssetIds.some(_assetId => _assetId === assetId)
+    const isUsdt = assetId === usdtAssetId
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/hooks/queries/useIsApprovalRequired.ts]: This one makes absolute sense! 5bb2fd73912ac612d4fafe7c06ed7052a51cd2f

File: src/hooks/queries/useIsApprovalRequired.ts (hunk 2 above)
Comment (0xApotheosis) [reply src/hooks/queries/useIsApprovalRequired.ts:15]: Yeah when you put it that way it makes complete sense, happy with this as-is 🤝

**Pull request merged**